packages/lighthouse-service/src/helpers.ts

```diff
@@ -90,6 +103,19 @@ class LHCIHelper {
                 throw new Error( error );
             } );
     }
+    
+    fetchProjectBranches ( serverBaseUrl: string, projectID: string) {
+        let headers = new Headers();
+        headers.append( `Content-Type`, `application/json` );
+        return fetch( `${ serverBaseUrl }/v1/projects/${ projectID }/branches`, {
+            method: 'GET',
+            headers: headers,
+            redirect: 'follow',
+        } ).then( ( res: any ) => res.json() )
```

Comment:
Handle http failures as fetch wont handle them.

Reply:
this is handled below